src/layers/ipv4.rs

```diff
@@ -89,22 +267,39 @@ impl Layer for IPv4 {
             });
         }
         self.tos = bytes[1];
-        self.len = (bytes[2] as u16) << 8 | (bytes[3] as u16);
-        self.id = (bytes[4] as u16) << 8 | (bytes[5] as u16);
-        let flags_offset = (bytes[6] as u16) << 8 | (bytes[7] as u16);
+        self.len = u16::from_be_bytes(bytes[2..4].try_into().unwrap());
```

Comment:
Any reasons why changed? Original idea was - the operation might be slightly faster than `from_bytes..try_into..unwrap`. But possible I am missing something.

Reply:
I was initially fixing a bug related to reading `self.frag_offset` while making this change, I didn't think a lot about performance, but it looks like its slightly better, or comparable, because of the one less bounds checking maybe? https://godbolt.org/z/rMhGjj6aM

Reply:
Yeah. Looks like there is no really big performance difference (which I was thinking would likely be the case), so yours is fine too!